### PR TITLE
Fix: check if legacy context force multishop mode

### LINKED ISSUE #36896: Carriers : Price Ranges are not displayed on edition (there ot not there, that's the question ?)
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Carriers : Price Ranges are not saved at creation

### Expected behavior

_No response_

### Steps to reproduce

1. Go to BO
2. Go to Carriers
3. Create a carrier
4. Add a range
image
5. Save. The carrier is created
6. Edit it
7. Check ranges : prices are empty
image

### PrestaShop version(s) where the bug happened

9.0.x

### PHP version(s) where the bug happened

8.1

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

@PrestaShopCorp

### PR DESCRIPTION
**Pull request by Nakahiru**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Since the new builder ShopContextBuilder.php was implemented to handle context, there were some issues regarding the legacy controller that forced the context to Shop::CONTEXT_ALL. For example, the controller AdminCarrierWizardController.php is set to Shop::CONTEXT_ALL and all stuff regarding carrier does not handle multishop. So all writing to the database regarding carrier is saved with id_shop and id_shop_group to NULL (like multishop). Since ShopContextBuilder.php, It was no longer the case, and the context was overridden by ShopContextBuilder.php to a single shop instead. Carrier were saved with id_shop.
| Type?             | bug fix
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | See in the linked issue #36896
| UI Tests          | https://github.com/Nakahiru/ga.tests.ui.pr/actions/runs/11212999910
| Fixed issue or discussion?     | Fixes #36896
| Related PRs       | none
| Sponsor company   | PrestaShop

### CODE REVIEW

**Comment by ps-jarvis** [comment]
Hi, thanks for this contribution!

I found some issues with the Pull Request description:

* The `type` should be one of these: `new feature`, `improvement`, `bug fix` or `refacto`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#branch))
* The `category` should be one of these: `FO`, `BO`, `CO`, `IN`, `WS`, `TE`, `LO`, `ME` or `PM`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#branch))

Would you mind having a look at it? This will help us understand how interesting your contribution is, thank you very much!

(Note: this is an automated message, but answering it will reach a real human)

**Review by uma1894603 — APPROVED** [review]

**Review by nicosomb — APPROVED** [review]

**Review by tleon — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]
Thanks @Nakahiru

**Review by florine2623 — APPROVED** [review]
Hello @Nakahiru ,

Auto tests ✅  : https://github.com/florine2623/testing_pr/actions/runs/11250356395

Tested the standard case, from an existing carrier, from one that I created.
Tested ranges according to price and weight.
Tested with one or multiple ranges.
Tested with one or multiple zones.

LGTM ✅ it is QA ✅ 
Thanks!

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**
